### ISSUE TITLE
Changes to the Extension API

### ISSUE THREAD

**Issue by melisgl** [melisgl/mgl-pax#23]
I'm going to overhaul the PAX::@EXTENSION-API because its semantics are rather baroque. If your code will be broken by the following proposed changes, I'd like to hear about it.

The changes will _not_ affect normal use (e.g. DEFSECTION, navigating, documenting, browsing, transcribing), or the obscure DEFINE-SYMBOL-LOCATIVE-TYPE, DEFINE-DEFINER-FOR-SYMBOL-LOCATIVE-TYPE, but the following changes are planned:

- REFERENCE will be renamed to something shorter.
- MAKE-REFERENCE may go away.
- The contract and maybe the name of CANONICAL-REFERENCE will change.
- The LOCATE-AND-* stuff will be removed.
- The contracts and names of LOCATE-OBJECT, DOCUMENT-OBJECT, DOCSTRING may change.

In summary, code that works with REFERENCES or defines a locative type (except by DEFINE-SYMBOL-LOCATIVE-TYPE) will break.

**Comment by melisgl** [comment]
This is a work in progress. The changes are going to be more extensive than described above. Dealing with references (to definitions) is going to be factored out into its own library similar to https://github.com/Shinmera/definitions, but which doesn't quite serve the needs of PAX.

**Comment by melisgl** [comment]
Here is the pull request: https://github.com/melisgl/mgl-pax/pull/29.
I'm going to give it a day or two before merging it and check that I'm not breaking anything in Quicklisp.
In the meantime, I'd appreciate any comments on the DRef api (see the readme):
https://github.com/melisgl/mgl-pax/tree/dref/dref

**Comment by melisgl** [comment]
This is committed as 171e93f2209147c9ab8db8ac79eeb2e911156259.
